subprojects/core/src/main/java/org/gradle/initialization/DefaultSettings.java

```diff
@@ -313,4 +321,14 @@ public SourceControl getSourceControl() {
     public void enableFeaturePreview(String name) {
         services.get(FeaturePreviews.class).enableFeature(name);
     }
+
```

Comment:
It would be good to improve the `Instantiator` implementation to mix some or all of this stuff in so we don't have to keep implementing this (and using internal types). It doesn't matter so much for `Settings` but for plugin types it does. We can do this in a later change, though.

Reply:
@oehme Agrees with this sentiment.